### PR TITLE
[KYUUBI #2752] Kyuubi Spark TPC-DS Connector - configurable catalog's name by initialize method

### LINKED ISSUE #2752: [Subtask] Kyuubi Spark TPC-DS Connector - configurable catalog's name by initialize method
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues) and found no similar issues.

### Describe the subtask

configurable catalog's name by initialize method

### Parent issue

https://github.com/apache/incubator-kyuubi/issues/2538

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### _Why are the changes needed?_

fix #2752 

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSCatalog.scala [hunk 1]
```diff
@@ -37,10 +37,13 @@ class TPCDSCatalog extends TableCatalog with SupportsNamespaces {
 
   var options: CaseInsensitiveStringMap = _
 
-  override def name: String = "tpcds"
+  var _name: Option[String] = None
```
Comment (pan3793) [inline extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSCatalog.scala:40]: var _name: String = _

Comment (jiaoqingbo) [reply extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSCatalog.scala:40]: ok

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Please rebase and resolve the conflicts

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**